### PR TITLE
chore: add action to automate creation of release branches/prs

### PR DESCRIPTION
**Pull request by Panaetius**

### CODE REVIEW

**Comment by rokroskar** [comment]
Great, this makes my release instructions easier 😆

File: .github/workflows/create-release-branch.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          fetch-depth: 0
+          token: "${{ secrets.RENKUBOT_GITHUB_TOKEN }}"
+          ref: '${{ github.event.inputs.target_branch}}'
+      - name: Create Branch
+        run: git checkout -b "release-${{ github.event.inputs.version }}"
+      - name: Update changelog
+        id: changelog
+        shell: bash
+        run: |
+          VERSION_LENGTH=$(echo -n "${{ github.event.inputs.version }}"|wc -m)
```
Comment (leafty) [inline .github/workflows/create-release-branch.yml:30]:
```suggestion
          VERSION_LENGTH=$(echo -n "${{ github.event.inputs.version }}" | wc -m)
```

File: .github/workflows/create-release-branch.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          DELIMITER=$(printf "%.0s-" $(seq $VERSION_LENGTH))
+          sed -i -E "s/^(.*.. _changelog:)/\1\n\n${{ github.event.inputs.version }}\n$DELIMITER\n\n/" CHANGELOG.rst
+          head -n 10 CHANGELOG.rst
+      - name: commit changes
```
Comment (leafty) [inline .github/workflows/create-release-branch.yml:34]:
```suggestion
      - name: Commit changes
```

File: .github/workflows/create-release-branch.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        run: |
+          git config --global --add user.name "Renku Bot"
+          git config --global --add user.email "renku@datascience.ch"
+          git config push.autoSetupRemote true
```
Comment (leafty) [inline .github/workflows/create-release-branch.yml:38]: Maybe separate this into a "Configure git" step?

**Review by leafty — COMMENTED** [review]
Some small nitpicks:

**Comment by Panaetius** [comment]
@leafty agreed, there's also some other things we could consider adding, like creating a draft GH release for the version. But I didn't want to make it too complicated to begin with.

**Pull request merged**